src/main/java/com/productdock/library/gateway/book/BookDetailsDto.java

```diff
@@ -0,0 +1,19 @@
+package com.productdock.library.gateway.book;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+import java.util.List;
+
+@Getter
+@Setter
+@AllArgsConstructor
+@NoArgsConstructor
+public class BookDetailsDto {
+    private Object bookDataDto;
+    private List<Object> rentalRecordsDto;
```

Comment:
Is there a reason to have a `*Dto` suffix for `bookData` and `rentalRecords`?
`BookDetailsDto` is a fine name for the class, but I guess there is no need for DTO in the attribute names.
Can BookDetailsDto be a `record` instead of a `class`?